synapse/__init__.py

```diff
@@ -29,4 +29,7 @@
 tmax = multiprocessing.cpu_count() * 8
 
 s_glob.plex = s_plex.Plex()
+s_glob.plex._fini_atexit = True
+
 s_glob.pool = s_threads.Pool(maxsize=tmax)
+s_glob.pool._fini_atexit = True
```

Comment:
unfortunately, the plex() fini is done as an async fini() inside of the thread that it is trying to shut down. It's a weird thing to teardown.